### ISSUE TITLE
Memory error in Raylib

### ISSUE THREAD

**Issue by predo2810** [SebLague/Chess-Challenge#75]
So, when i try to run the application, a console window open, and, after about 1 second of loading, this show up:

![image](https://github.com/SebLague/Chess-Challenge/assets/93801931/77878799-7f43-4f91-aa8b-dbe502fdf966)

I don't really know what's happening, and, since the only modifiable file is the MyBot.cs, i don't it would be fair for me to try to fix this myself...

I'm using visual studio 2022, i installed the C# (.NET) development and the desktop development with C# (I don't I'm missing something)

![image](https://github.com/SebLague/Chess-Challenge/assets/93801931/02894d7a-d3fa-45bc-904c-2c39e1e1298d)
(it's in Portuguese and it says:
Desktop development with .NET (left one) and
Universal Windows platform development (right one))

**Comment by zment4** [comment]
I have the same. I'll try to figure out a workaround but I have really no idea what's going on. And yes on Visual Studio 2022.

**Comment by zment4** [comment]
If I set SDF_ENABLED I get the same memory error but from LoadFontEx... 🤔

**Comment by predo2810** [comment]
@zment4 something is definitely wrong XD
You're using Windows 11?

**Comment by zment4** [comment]
Windows 10 Home

su 23. heinäk. 2023 klo 4.26 EverRak ***@***.***> kirjoitti:

**Comment by predo2810** [comment]
@zment4 ok... so the problem is not windows

**Comment by zment4** [comment]
I tested with a base raylib project, and it fails on creating a GL window. My current thesis on the problem is that my laptop's GPU is not supported by raylib 😢

**Comment by predo2810** [comment]
oh no... That's probably my problem too... i don't really have a GPU so...

**Comment by zment4** [comment]
Talking with people on raylib discord we found the problem was OpenGL version support. Raylib defaults to OpenGL 3.3, but my laptop GPU doesn't support that high. Raylib can be compiled with an earlier version, so they made a test for OpenGL 2.1 and that worked just fine. I'm currently in the process of figuring out how to do it on my own so I can maybe share that if I get it working.

**Comment by predo2810** [comment]
@zment4 Yeaaah! Thank you dude :D

**Comment by zment4** [comment]
Alright, managed to finally test the project with a Raylib DLL built for OpenGL 2.1. After building, the raylib.dll needs to be replaced with this one at location `Chess-Challenge\bin\Release\net6.0\runtimes\win-x64\native` and in case the problem was no support for OpenGL 3.3, this should make it run now. 

If I understand the build system correctly, the raylib.dll will get replaced after every build, so it's a bit of a chore, but I'll try to figure out a way to either make the replacing automatic (post-build copy action maybe?) or replace the DLL where it gets copied from in the first place.

[raylib_ogl21.zip](https://github.com/SebLague/Chess-Challenge/files/12138582/raylib_ogl21.zip)

**Comment by predo2810** [comment]
@zment4 Perfect, do you want to let the issue open?

**Comment by zment4** [comment]
I don't mind either way. Seeing as this more a workaround than a fix it might as well be left open. 

Oh and btw setting up a post-build step for copying the dll to the correct folder like so `copy /Y $(ProjectDir)..\raylib_ogl21.dll $(OutDir)runtimes\win-x64\native\raylib.dll` made building and running the program so much easier.

**Comment by predo2810** [comment]
> I don't mind either way. Seeing as this is more a workaround than a fix it might as well be left open. 
> 
> Oh and btw setting up a post-build step for copying the dll to the correct folder like so `copy /Y $(ProjectDir)..\raylib_ogl21.dll $(OutDir)runtimes\win-x64\native\raylib.dll` made building and running the program so much easier.

So, you found a "usable" fix for now?
Until now i made nothing except for thinking about what to do because i couldn't run the project :,)

**Comment by zment4** [comment]
Yes, replacing the Raylib-cs provided raylib.dll (which use OpenGL 3.3) with one that was compiled targeting OpenGL 2.1 (the one I linked) fixes the issue for me. With the post-build step copy it's also pretty seamless, after setting it up it works after every build without unnecessary manual copying.

**Comment by predo2810** [comment]
I will test it to see if i get something to work

**Comment by predo2810** [comment]
@zment4 - Well, it seems like my computer is worst than i expected... After doing what you said, and even after that, fixing some more things, i still didn't get it to work...

**Comment by Thomas-Dvorak** [comment]
I'm also getting the same error. I tried replacing the .dll file after building and that still didn't work for me. Also, for some reason I don't have the `Chess-Challenge\bin\Release\net6.0\runtimes\win-x64\native` directory. Instead of 'Release', it's 'Debug'.

**Comment by Thomas-Dvorak** [comment]
I'm building the project and replacing the file, and when I run it, it gives me the same error. It looks like it's re-building itself every every time it's run. I looked at the file after I replaced it (showing that the modified time was just a few minutes ago), and when I run it, the file (raylib.dll) is suddenly replaced with an older version (from 2023) which is probably the original. 
**Is there a fix for this?**
*I'm using Windows 10, but have a Core i3, which I don't know if it's causing the problem or not.*

**Comment by Devo-The-Developer** [comment]
I looked further into (at least my) the dotnet command lines and used 
`dotnet run Chess-Challenge.csproj --no-build` to run the project and it worked without error. You still have to replace the "raylib.dll" file every time, but that can be fixed from one of the comments from @zment4 .
